database-commons/src/main/java/io/cdap/plugin/db/batch/config/AbstractDBSpecificSinkConfig.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.batch.config;
+
+import com.google.common.collect.Maps;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.plugin.common.Constants;
+import io.cdap.plugin.db.ConnectionConfig;
+import io.cdap.plugin.db.connector.AbstractDBSpecificConnectorConfig;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Abstract Config for DB Specific Sink plugin
+ */
+public abstract class AbstractDBSpecificSinkConfig extends PluginConfig implements DatabaseSinkConfig {
+  public static final String TABLE_NAME = "tableName";
+  public static final String TRANSACTION_ISOLATION_LEVEL = "transactionIsolationLevel";
+
+  @Name(Constants.Reference.REFERENCE_NAME)
+  @Description(Constants.Reference.REFERENCE_NAME_DESCRIPTION)
+  private String referenceName;
+
+  @Name(TABLE_NAME)
+  @Description("Name of the database table to write to.")
+  @Macro
+  private String tableName;
+
+  public String getTableName() {
+    return tableName;
+  }
+
+  /**
+   * Adds escape characters (back quotes, double quotes, etc.) to the table name for
+   * databases with case-sensitive identifiers.
+   *
+   * @return tableName with leading and trailing escape characters appended.
+   * Default implementation returns unchanged table name string.
+   */
+  public String getEscapedTableName() {
+    return tableName;
+  }
+
+  public boolean connectionParamsContainsMacro() {
+    return containsMacro(TABLE_NAME) || containsMacro(ConnectionConfig.DATABASE)
+      || getConnection().connectionParamsContainsMacro();
+  }
+
+  @Nullable
+  public String getTransactionIsolationLevel() {
+    return null;
```

Comment:
this just return null and I don't see any field variable related to it, do we still need this method?

Reply:
Yeah, we need a default implementation of this method because its being called in the base class AbstractDbSink,
https://github.com/data-integrations/database-plugins/blob/873f86ad337403dadd824607d692ba3524d2dbce/database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java#L157.

Reply:
Please mark this method with `@Override`